### PR TITLE
Shared feed import

### PR DESCRIPTION
**Pull request by caiosba**
## Description

This PR adds a new button to the shared feed item page: "Import Media". Media can be imported as one of three different types:

- [x] Create new item
- [x] Add to cluster exemplar
- [x] Search for item

See more details specs and Figma designs in reference ticket for more details.

Until the backend for this work is merged, it requires the Check API branch `feature/CV2-4109-shared-feed-import`.

Reference: CV2-4109.

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Manually, and unit tests added for new components.

## Things to pay attention

@amoedoamorim , please review localization for sentences that have links inside them.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [x] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Comment by caiosba** [comment]
FYI @brianfleming (and reviewers): the only thing different from the design is the success toast:

Captura de tela de 2024-04-12 06-42-56

I didn't see a way to do that with our current `setFlashMessage` component. Maybe we can work on that in a separate ticket, and make it accept an action button or something like that.

Everything else should be implemented according to the designs.

**Comment by brianfleming** [comment]
> FYI @brianfleming (and reviewers): the only thing different from the design is the success toast:
> 
> Captura de tela de 2024-04-12 06-42-56
> 
> I didn't see a way to do that with our current `setFlashMessage` component. Maybe we can work on that in a separate ticket, and make it accept an action button or something like that.
> 
> Everything else should be implemented according to the designs.

Not a problem @caiosba There is a ticket out there to replace our FlashMessage with the CDS version using our current `<Alert... `component, but we haven't done it yet. As long as this work uses the current flash message so the user gets the feedback, we are good for now.

**Comment by caiosba** [comment]
Thanks, that's what I did! Glad to know there is already a ticket for it.

File: src/app/components/feed/FeedImportDialog.js [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+          <div>
+            <FormattedMessage
+              tagName="h6"
+              id="feedImportDialog.title"
+              defaultMessage="Import {mediaCount} media into your workspace"
+              description="Title for the import media dialog on the feed item page."
+              values={{
+                mediaCount: cluster.media_count,
+              }}
+            />
```
[the remark this answers is not in the document]
Comment (brianfleming) [reply src/app/components/feed/FeedImportDialog.js:123]: Also note when updating this, we want to go to `media` for both singular and plural in English. Need to have this code set up for pluralized for other languages, but just wanted to call out the stupid English exception

Additionally, the translators called our attention to this article, with the following note:

> Transifex recommends to use the same variable for both the singular and plural form.

https://help.transifex.com/en/articles/6231958-working-with-plurals-and-genders

File: src/app/components/feed/FeedImportDialog.js [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+                          defaultMessage="existing item to import"
+                          description="This is the text of the link to create a new item in the feed import dialog. It's used in the middle of a sentence like: You can import this media in a new item or search for an 'existing item to import'."
+                        />
+                      </span>
+                    ),
+                  }}
+                />
+              </div>
+            </div>
+          )}
```
[the remark this answers is not in the document]
Comment (amoedoamorim) [reply src/app/components/feed/FeedImportDialog.js:257]: Looking forward to upgrading react-intl:

https://formatjs.io/docs/react-intl/components/#rich-text-formatting

File: src/app/components/feed/FeedImportDialog.js [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+            onClick={onClose}
+            disabled={saving}
+            label={
+              <FormattedMessage
+                id="feedImportDialog.cancel"
+                defaultMessage="Cancel"
+                description="Label of a button to close the import dialog window on the feed item page."
+              />
+            }
+          />
```
Comment (amoedoamorim) [inline src/app/components/feed/FeedImportDialog.js:282]: Please reuse same props as global.edit in globalStrings.js

**Review by amoedoamorim — CHANGES_REQUESTED** [review]
Just a couple of minor copy issues fixes requested

File: src/app/components/feed/FeedImportDialog.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (caiosba) [reply src/app/components/feed/FeedImportDialog.js:123]: Done!

File: src/app/components/feed/FeedImportDialog.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (caiosba) [reply src/app/components/feed/FeedImportDialog.js:257]: Yeah, also looking forward to it: https://github.com/meedan/check-web/blob/develop/src/app/components/feed/FeedClusters.js#L315

File: src/app/components/feed/FeedImportDialog.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (amoedoamorim) [reply src/app/components/feed/FeedImportDialog.js:123]: @caiosba did you forget to push your changes? I don't see them

[the remark this answers is not in the document]
Comment (caiosba) [reply src/app/components/feed/FeedImportDialog.js:123]: Yes, sorry, I'm answering as I do them locally but I haven't pushed yet... I'll re-request review when I finish

File: src/app/components/feed/FeedImportDialog.js (hunk 3 above)
Comment (caiosba) [reply src/app/components/feed/FeedImportDialog.js:282]: Done!

**Review by amoedoamorim — APPROVED** [review]
:lgtm:

**Pull request merged**